src/tests/ef_vi/eflatency.c

```diff
@@ -214,24 +222,40 @@ generic_ping(struct eflatency_vi* rx_vi, struct eflatency_vi* tx_vi,
 {
   struct timeval start, end;
   int i;
-  int do_rx_post = ( rx_vi->vi.nic_type.arch != EF_VI_ARCH_EFCT );
+  const int do_rx_post = ( rx_vi->vi.nic_type.arch != EF_VI_ARCH_EFCT );
+  uint8_t * const tx_pkt = pkt_bufs[FIRST_TX_BUF]->dma_buf;
+  uint8_t pattern = 0;
+
+  for( i = 0; i < cfg_warmups; ++i, pattern++ ) {
+    memset(&tx_pkt[HEADER_SIZE], pattern, cfg_payload_len);
+    checksum_udp_pkt(tx_pkt);
 
-  for( i = 0; i < cfg_warmups; ++i ) {
     tx_send(tx_vi);
     if( do_rx_post )
       rx_post(&rx_vi->vi);
     rx_wait(rx_vi);
-   generic_desc_check(tx_vi, 0);
+    generic_desc_check(tx_vi, 0);
   }
 
   gettimeofday(&start, NULL);
 
-  for( i = 0; i < cfg_iter; ++i ) {
+  for( i = 0; i < cfg_iter; ++i, pattern++ ) {
+    memset(&tx_pkt[HEADER_SIZE], pattern, cfg_payload_len);
+    checksum_udp_pkt(tx_pkt);
+
     uint64_t start = ci_frc64_get();
     tx_send(tx_vi);
     if( do_rx_post )
       rx_post(&rx_vi->vi);
     rx_wait(rx_vi);
+
+    if (cfg_validate && cfg_payload_len > 0)
+    {
+      const uint8_t rx_pattern = rx_vi->rx_pkt[HEADER_SIZE];
+      if (pattern != rx_pattern)
```

Comment:
I wonder if when validating it's worth validating the whole packet using `memcp` for example, or if for more detail about the first octet that's different a custom loop would suffice.

Reply:
The first version of the patch (as you noted above) only set the first byte... I can add a loop to check the rest of them.